src/LibraryViewExtension/LibraryViewCustomization.cs

```
@@ -0,0 +1,126 @@
+        /// elements.
+        /// </summary>
+        public bool AddElements(IEnumerable<LayoutElement> elements, string sectionText = "")
+        {
+            var text = DefaultSectionName;
+            if (!string.IsNullOrEmpty(sectionText))
+            {
+                text = sectionText;
+            }
+            var spec = GetSpecification();
+            var section = spec.sections.FirstOrDefault(x => string.Equals(x.text, text));
+            if (section == null)
+            {
+                section = new LayoutSection(text);
+                spec.sections.Add(section);
+            }
+            section.childElements.AddRange(elements);
+            int count = section.childElements.Count;
+            if (section.childElements.GroupBy(e => e.text).Count() < count) return false;
+
+            //Update the specification
+            SetSpecification(spec);
+            return true;
+        }
+
+        /// <summary>
+        /// Allows clients to add a list of include path to a given section. This 
+        /// operation will fail if the path property of any of the includes 
+        /// of the given section conflicts with the path property of the given 
+        /// includes.
+        /// </summary>
+        public bool AddIncludeInfo(IEnumerable<LayoutIncludeInfo> includes, string sectionText = "")
+        {
+            var text = DefaultSectionName;
+            if (!string.IsNullOrEmpty(sectionText))
+            {
+                text = sectionText;
+            }
+            var spec = GetSpecification();
+            var section = spec.sections.FirstOrDefault(x => string.Equals(x.text, text));
+            if (section == null)
+            {
+                section = new LayoutSection(text);
+                spec.sections.Add(section);
+            }
+            section.include.AddRange(includes);
+            int count = section.include.Count;
+            if (section.include.GroupBy(e => e.path).Count() < count) return false;
```

Yes, it's a kind of user error, we can either throw an exception or do a console log.